### PR TITLE
refactor: support distinct JSON format and improve type conversions

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

support distinct JSON format and improve type conversions

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

## Release Notes

- **New Features**
  - Introduced enhanced handling for JSON data types, including new methods for converting and parsing JSON values.
  - Added a method to `ConcreteDataType` for casting to `JsonType`.

- **Bug Fixes**
  - Improved error handling during JSON type conversions and parsing, providing clearer error messages.

- **Improvements**
  - Enhanced compatibility between JSON and binary types in data validation.
  - Streamlined JSON data handling in MySQL and PostgreSQL writers for better performance and error management.

### CODE REVIEW

**Comment by WenyXu** [comment]
@coderabbitai summary

**Review by CookiePieWw — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
Rest LGTM

File: src/datatypes/src/types/json_type.rs [hunk 1]
```diff
@@ -65,3 +79,19 @@ impl DataType for JsonType {
         }
     }
 }
+
+/// Converts a json type value to string
+pub fn json_type_value_to_string(val: &[u8], format: &JsonFormat) -> Result<String> {
```
[the remark this answers is not in the document]
Comment (WenyXu) [reply src/datatypes/src/types/json_type.rs:84]: Keep the same style with `vector_type_value_to_string`

[the remark this answers is not in the document]
Comment (fengjiachun) [reply src/datatypes/src/types/json_type.rs:84]: ok

**Pull request merged**